packages/react/scroll-area/src/ScrollArea.tsx

```diff
@@ -663,52 +656,55 @@ const ScrollAreaScrollbarImpl = React.forwardRef((props, forwardedRef) => {
     const handleWheel = (event: WheelEvent) => {
       const element = event.target as HTMLElement;
       const isScrollbarWheel = scrollbar?.contains(element);
-      if (isScrollbarWheel) handleWheelScroll(event);
+      if (isScrollbarWheel) handleWheelScroll(event, maxScrollPos);
     };
     document.addEventListener('wheel', handleWheel, { passive: false });
     return () => document.removeEventListener('wheel', handleWheel, { passive: false } as any);
-  }, [viewport, scrollbar, handleWheelScroll]);
+  }, [viewport, scrollbar, maxScrollPos, handleWheelScroll]);
 
-  function handleDragScroll(event: React.PointerEvent<HTMLElement>) {
-    if (rectRef.current) {
-      const x = event.clientX - rectRef.current.left;
-      const y = event.clientY - rectRef.current.top;
-      onDragScroll({ x, y });
-    }
-  }
+  /**
+   * Update thumb position on sizes change
+   */
+  React.useEffect(handleThumbPositionChange, [sizes, handleThumbPositionChange]);
```

Comment:
This means it will run this everytime the reference to sizes changes right? but because it's an object that will be all the time no? I guess that's ok because you update them all in one go always right?

Reply:
It won't be all the time because `sizes` prop is coming from state in the parent component. I realise that isn't particularly clear though because it's a prop in this context ..... 🤔  hmmm.  Any ideas here?

Reply:
Yeah exactly, I basically had that question at first, then I hunted the code down to find that it was a state and that we were setting it all in one go. Not sure if we could improve it, `sizesState` may explain a bit, but honestly that'd be a PITA…